src/txn/txn_recover.c

```diff
@@ -430,6 +433,14 @@ __recovery_setup_file(WT_RECOVERY *r, const char *uri, const char *config)
           uri, r->files[fileid].uri, fileid);
     WT_RET(__wt_strdup(r->session, uri, &r->files[fileid].uri));
     WT_RET(__wt_config_getones(r->session, config, "checkpoint_lsn", &cval));
+
+    if ((ret = __wt_ckpt_last(r->session, config, &ckpt)) == 0) {
+        S2C(r->session)
+          ->max_write_gen = S2C(r->session)->base_write_gen =
+          WT_MAX(S2C(r->session)->max_write_gen, ckpt.write_gen + 1);
```

Comment:
Same comments here. Use `conn` and set in alphabetical order. Can you refactor so that you don't have the same code twice, here and in `wt_meta_set_base_write_gen`?

Why is this code located here exactly? This is in between getting the `checkpoint_lsn` config value and then using that LSN value we got. I think if this is necessary, it should be before or after all the LSN related work (which is most of this function).

Reply:
I don't know much about this part of the code. I'm still in the stage of exploring possible solutions. Will discuss this today.